### PR TITLE
[BugFix] Resolve bugs in computer_vision_fine_tuning.py example

### LINKED ISSUE #5906: BaseFinetuning Callback freeze_before_training is never called
## 🐛 Bug

I think there might be a bug in the implementation of the `BaseFinetuning` callback, and in particular in the following lines:
https://github.com/PyTorchLightning/pytorch-lightning/blob/a028171f26d781491fa87fd27adf0e6dc1d7c3ae/pytorch_lightning/callbacks/finetuning.py#L236-L237

`freeze_before_training` is meant to be over-riden by child classes so that the freezing logic is implemented before the start of training. However, the call to this method happens using the `on_before_accelerator_backend_setup` hook. I am not sure where this hook has come from as I was unable to find it in the `latest` version of the [documentation](https://pytorch-lightning.readthedocs.io/en/latest/callbacks.html#available-callback-hooks):

Testing the callback using the `MilestonesFinetuningCallback` from the related [example](https://github.com/PyTorchLightning/pytorch-lightning/blob/a028171f26d781491fa87fd27adf0e6dc1d7c3ae/pl_examples/domain_templates/computer_vision_fine_tuning.py#L65) resulted in the layers not being freezed before training. 

To make this work, I had to change the callback hook that `freeze_before_training` is called to `on_pretrain_routine_start`.

Apologies I wasn't able to provide a piece of code to reproduce this issue.

**On a separate note**, the `MilestonesFinetuningCallback` in the example uses wrong signatures:
https://github.com/PyTorchLightning/pytorch-lightning/blob/a028171f26d781491fa87fd27adf0e6dc1d7c3ae/pl_examples/domain_templates/computer_vision_fine_tuning.py#L72
here `module` should be `modules` to match the signature of the `freeze` method of the parent class:
https://github.com/PyTorchLightning/pytorch-lightning/blob/a028171f26d781491fa87fd27adf0e6dc1d7c3ae/pytorch_lightning/callbacks/finetuning.py#L147

Same applies to `unfreeze_and_add_param_group`:
https://github.com/PyTorchLightning/pytorch-lightning/blob/a028171f26d781491fa87fd27adf0e6dc1d7c3ae/pl_examples/domain_templates/computer_vision_fine_tuning.py#L77-L79
and
https://github.com/PyTorchLightning/pytorch-lightning/blob/a028171f26d781491fa87fd27adf0e6dc1d7c3ae/pl_examples/domain_templates/computer_vision_fine_tuning.py#L83-L84
Compare to:
https://github.com/PyTorchLightning/pytorch-lightning/blob/a028171f26d781491fa87fd27adf0e6dc1d7c3ae/pytorch_lightning/callbacks/finetuning.py#L197-L198

## Please reproduce using the BoringModel

Bug is in branch `release/1.2-dev` so I wouldn't be able to reproduce using  `pip` installation.

## Environment

 - PyTorch Version (e.g., 1.0): 1.6.0+cu101
 - OS (e.g., Linux): Linux
 - How you installed PyTorch (`conda`, `pip`, source): pip
 - Build command you used (if compiling from source):
 - Python version: 3.7.7
 - CUDA/cuDNN version: 10.1
 - GPU models and configuration:  4 X GeForce GTX 1080 Ti
 - Any other relevant information:

### Additional context

Happy to submit a PR to fix these issues.

### PR DESCRIPTION
**Pull request by tchaton**
## What does this PR do?

This PR resolves some typos and add a DataModule to the example for best pratices.

Fixes #5906

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified
 - [x] **Check that target branch and milestone match!**

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by carmocca — APPROVED** [review]
LGTM

**Review by rohitgr7 — APPROVED** [review]

**Review by Borda — APPROVED** [review]
